src/Integration.UnitTests/Connection/ConnectControllerTests.cs

```diff
@@ -0,0 +1,373 @@
+// <copyright file="ConnectControllerTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.Connection;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.TeamExplorer;
+using SonarLint.VisualStudio.Integration.WPF;
+using SonarLint.VisualStudio.Progress.Controller;
+using System;
+using System.Globalization;
+using System.Windows.Threading;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Connection
+{
+    [TestClass]
+    public class ConnectControllerTests
+    {
+        private ConfigurableHost host;
+        private ConfigurableSonarQubeServiceWrapper sonarQubeService;
+        private ConfigurableConnectionWorkflow connectionWorkflow;
+        private ConfigurableConnectionInformationProvider connectionProvider;
+        private ConfigurableServiceProvider serviceProvider;
+        private ConfigurableVsGeneralOutputWindowPane outputWindowPane;
+        private ConfigurableIntegrationSettings settings;
+
+        [TestInitialize]
+        public void TestInit()
+        {
+            this.sonarQubeService = new ConfigurableSonarQubeServiceWrapper();
+            this.connectionWorkflow = new ConfigurableConnectionWorkflow(this.sonarQubeService);
+            this.connectionProvider = new ConfigurableConnectionInformationProvider();
+            this.serviceProvider = new ConfigurableServiceProvider();
+            this.outputWindowPane = new ConfigurableVsGeneralOutputWindowPane();
+            this.serviceProvider.RegisterService(typeof(SVsGeneralOutputWindowPane), this.outputWindowPane);
+            this.settings = new ConfigurableIntegrationSettings();
+            this.host = new ConfigurableHost(this.serviceProvider, Dispatcher.CurrentDispatcher);
+            this.host.SonarQubeService = this.sonarQubeService;
+
+            var mefExports = MefTestHelpers.CreateExport<IIntegrationSettings>(settings);
+            var mefModel = ConfigurableComponentModel.CreateWithExports(mefExports);
+            this.serviceProvider.RegisterService(typeof(SComponentModel), mefModel);
+        }
+
+        #region  Tests
+        [TestMethod]
+        public void ConnectionController_Ctor_ArgumentChecks()
+        {
+            ConnectionController suppressAnalysisIssue;
+            Exceptions.Expect<ArgumentNullException>(() =>
+            {
+                suppressAnalysisIssue = new ConnectionController(null);
```

Comment:
Could also avoid the analysis issue by replacing the code block with an expression (`() => new ConnectionController(null)`). I was confused at first seeing this variable.